PancakeSwap-Bots/Prediction-Admin-Operatons/src/agent.spec.ts

```diff
@@ -0,0 +1,221 @@
+import { FindingType, FindingSeverity, Finding, HandleTransaction, TransactionEvent } from "forta-agent";
+import { BigNumber } from "ethers";
+import { provideHandleTransaction } from "./agent";
+import { createAddress, TestTransactionEvent } from "forta-agent-tools/lib/tests";
+import { Interface } from "@ethersproject/abi";
+import abi from "./abi";
+
+const pause = (time: BigNumber): Finding =>
+  Finding.fromObject({
+    name: "PancakePredictionV2 Operations",
+    description: "Pause event emitted",
+    severity: FindingSeverity.Info,
+    type: FindingType.Info,
+    alertId: "CAKE-9-1",
+    metadata: {
+      time: `${time}`,
+    },
+  });
+
+const unpause = (time: BigNumber): Finding =>
+  Finding.fromObject({
+    name: "PancakePredictionV2 Operations",
+    description: "Unpause event emitted",
+    severity: FindingSeverity.Info,
+    type: FindingType.Info,
+    alertId: "CAKE-9-2",
+    metadata: {
+      time: `${time}`,
+    },
+  });
+
+const newOperatorAddress = (address: string): Finding =>
+  Finding.fromObject({
+    name: "PancakePredictionV2 Operations",
+    description: "NewOperatorAddress event emitted",
+    severity: FindingSeverity.Info,
+    type: FindingType.Info,
+    alertId: "CAKE-9-3",
+    metadata: {
+      address,
+    },
+  });
+
+const newAdminAddress = (address: string): Finding =>
+  Finding.fromObject({
+    name: "PancakePredictionV2 Operations",
+    description: "NewAdminAddress event emitted",
+    severity: FindingSeverity.Info,
+    type: FindingType.Info,
+    alertId: "CAKE-9-4",
+    metadata: {
+      address,
+    },
+  });
+
+const newOracle = (address: string): Finding =>
+  Finding.fromObject({
+    name: "PancakePredictionV2 Operations",
+    description: "NewOracle event emitted",
+    severity: FindingSeverity.Info,
+    type: FindingType.Info,
+    alertId: "CAKE-9-5",
+    metadata: {
+      address,
+    },
+  });
+
+const newTreasuryFee = (time: BigNumber, fee: BigNumber): Finding =>
+  Finding.fromObject({
+    name: "PancakePredictionV2 Operations",
+    description: "NewTreasuryFee event emitted",
+    severity: FindingSeverity.Info,
+    type: FindingType.Info,
+    alertId: "CAKE-9-6",
+    metadata: {
+      time: `${time}`,
+      fee: `${fee}`,
+    },
+  });
+
+describe("PancakePredictionV2-Operations agent tests suite", () => {
+  const iface = new Interface(abi.CAKE);
+
+  it("should ignore empty txns", async () => {
+    const cake: string = createAddress("0x01");
+    const handler: HandleTransaction = provideHandleTransaction(cake);
```

Comment:
I would suggest creating a `mockInterface` for this inside the `.spec.ts` so that the `abi.ts` stays "clean" (you then can create even a completely fake event, not existing in the real contract)

Reply:
Thanks for the suggestions, I did exactly that! `abi.ts` untouched.